legends/load_legends.py

```diff
@@ -0,0 +1,309 @@
+                    if ds.meta["width"] > MAX_LENGTH or ds.meta["height"] > MAX_LENGTH:
+                        blocks_data = []
+                        # filter out nodata values, flatten the output to 1d array and concatenate lists
+                        for _, window_read in ds.block_windows(1):
+                            block = ds.read(window=window_read)
+                            if type(ds.nodata) == float:
+                                block = block.astype(float)
+                            block[block == ds.nodata] = np.nan
+                            block = block[~np.isnan(block)]
+                            blocks_data.extend(block)
+
+                    # read the whole raster (larger memory footprint, but faster - less IO operations)
+                    else:
+                        # filter out nodata values and flatten the output to 1d array
+                        blocks_data = ds.read(1)
+                        if type(ds.nodata) == float:
+                            blocks_data = blocks_data.astype(float)
+                        blocks_data[blocks_data == ds.nodata] = np.nan
+                        blocks_data = blocks_data[~np.isnan(blocks_data)]
+
+                    # adapt the sample space to the size of the raster, then sample using scrambled Sobol sequences (use multiples of 2)
+                    if len(total_sample_data) < MAX_SIZE:
+                        l_bounds = [0]
+                        u_bounds = [len(blocks_data)]
+                        sample_data = np.empty(0)
+                        m_exp = np.ceil(math.log(SAMPLE_SIZE, 2)).astype(int)
+                        sampler = qmc.Sobol(d=1, scramble=True, seed=1)
+                        sample = sampler.random_base2(m_exp)
+                        sample = qmc.scale(sample, l_bounds, u_bounds)
+                        sample = np.rint(sample).astype(int)
+                        for sample_coords in sample:
+                            index = sample_coords[0]
+                            val = blocks_data[index]
+                            sample_data = np.append(sample_data, val)
+                        total_sample_data = np.append(total_sample_data, sample_data)
+
+                    # compute min and max of the whole layer
+                    breaks_min = min(breaks_min, np.min(blocks_data))
+                    breaks_max = max(breaks_max, np.max(blocks_data))
+
+            # computing breaks, considering all rasters
+            breaks = np.percentile(total_sample_data, percentiles_list)
+
+        else:
+            print(
+                'Computing legend breaks - dataset {}, layer "{}"\n'.format(
+                    ds_id, variable
+                )
+            )
+            vector = gdf
+            vector["variable"] = vector["variables"].apply(
+                lambda x: list(x.values())[0]
+            )
+            data = np.array(vector["variable"])
```

Comment:
For dataset 3, the ValueError("The breaks are less than 2") is raised.
I would suggest using this here:
`data = np.array(vector["variable"].dropna())`